src/main/java/team/ebi/epicbanitem/configuration/Settings.java

```diff
@@ -166,9 +175,11 @@ private void onPostInit(GamePostInitializationEvent event) throws IOException {
         configFile.save();
     }
 
-    @Listener
-    public void onStarted(GameStartedServerEvent event) {
+    private void onStarted(GameStartedServerEvent event) {
         NbtTagDataUtil.printToLogger(logger::debug, this.printItemToBlockMapping());
         logger.debug("Change the value of 'print-item-to-block-mapping' to enable or disable detailed output.");
+        if (checkUpdate) {
+            CompletableFuture.runAsync(()->injector.getInstance(UpdateChecker.class).checkUpdate());
```

Comment:
This lambda will cause the `UpdateChecker` loaded asynchronously (not initialized in the main thread). Maybe `injector.getInstance(UpdateChecker.class)::checkUpdate` is a better choice.